opsmngr/backup_administrator.go

```diff
@@ -0,0 +1,874 @@
+type SyncService interface {
+	ListSyncs(context.Context, *atlas.ListOptions) (*Syncs, *atlas.Response, error)
+	GetSync(context.Context, string) (*Sync, *atlas.Response, error)
+	CreateSync(context.Context, *Sync) (*Sync, *atlas.Response, error)
+	UpdateSync(context.Context, string, *Sync) (*Sync, *atlas.Response, error)
+	DeleteSync(context.Context, string) (*atlas.Response, error)
+}
+
+// DaemonService is an interface for using the Backup Daemon
+// endpoints of the MongoDB Ops Manager API.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/admin/backup/backup-daemon-config/
+type DaemonService interface {
+	ListDaemons(context.Context, *atlas.ListOptions) (*Daemons, *atlas.Response, error)
+	GetDaemon(context.Context, string) (*Daemon, *atlas.Response, error)
+	UpdateDaemon(context.Context, string, *Daemon) (*Daemon, *atlas.Response, error)
+	DeleteDaemon(context.Context, string) (*atlas.Response, error)
+}
+
+// ProjectJobService is an interface for using the Project Job
+// endpoints of the MongoDB Ops Manager API.
+//
+// See more: https://docs.opsmanager.mongodb.com/current/reference/api/admin/backup/backup-group-config/
+type ProjectJobService interface {
+	ListProjectJobs(context.Context, *atlas.ListOptions) (*ProjectJobs, *atlas.Response, error)
+	GetProjectJob(context.Context, string) (*ProjectJob, *atlas.Response, error)
+	UpdateProjectJob(context.Context, string, *ProjectJob) (*ProjectJob, *atlas.Response, error)
+}
+
+// BackupConfigsServiceOp provides an implementation of the BackupConfigsService interface
+type BackupAdministratorServiceOp service
+
+var _ BackupAdministratorService = &BackupAdministratorServiceOp{}
+
+// AdminConfig contains the common fields of backup administrator structs
+type AdminConfig struct {
+	ID                   string   `json:"id,omitempty"`
+	URI                  string   `json:"uri,omitempty"`
+	WriteConcern         string   `json:"writeConcern,omitempty"`
+	Labels               []string `json:"labels,omitempty"`
+	SSL                  bool     `json:"ssl"`
+	AssignmentEnabled    bool     `json:"assignmentEnabled"`
+	EncryptedCredentials bool     `json:"encryptedCredentials"`
+	UsedSize             int64    `json:"usedSize,omitempty"`
+}
+
+// S3Blockstore represents a S3Blockstore in the MongoDB Ops Manager API
+type S3Blockstore struct {
+	Blockstore
+	AWSAccessKey           string `json:"awsAccessKey,omitempty"`
+	AWSSecretKey           string `json:"awsSecretKey,omitempty"`
+	DisableProxyS3         string `json:"disableProxyS3,omitempty"`
+	S3AuthMethod           string `json:"s3AuthMethod,omitempty"`
+	S3BucketEndpoint       string `json:"s3BucketEndpoint,omitempty"`
+	S3BucketName           string `json:"s3BucketName,omitempty"`
+	S3MaxConnections       int64  `json:"s3MaxConnections,omitempty"`
+	AcceptedTos            bool   `json:"acceptedTos"`
+	SSEEnabled             bool   `json:"sseEnabled"`
```

Comment:
similar to above, this can be ommited on the updated, which makes me flag from the previous PR that those booleans should also be pointers and omit empty
See: https://github.com/mongodb/go-client-mongodb-ops-manager/blob/b15016909b60e6324f05a18dde51f99644417771/opsmngr/backup_configs.go